server/admin/http/handler.go

```diff
@@ -373,13 +373,23 @@ func batchIndexerEntries(batchSize int, putChan <-chan multihash.Multihash, valu
 // ----- admin handlers -----
 
 func (h *adminHandler) healthCheckHandler(w http.ResponseWriter, r *http.Request) {
-	w.WriteHeader(http.StatusOK)
-	// TODO: Report on indexer core health?
-	_, err := w.Write([]byte("\"OK\""))
+	healthStatus := &HealthStatus{Server: "OK", ValueStore: "OK"}
```

Comment:
The HealthStatus contains two status values, yet the code only ever sets one or the other. So, in the case of an error, it seems sufficient to return only a single error that describes the failure.  Also, if there is no error, then returning a 200 should be sufficient to declare that there is no error, and returning `HealthStatus{Server: "OK", ValueStore: "OK"}` is not necessary.

When `json.Marshal` fails the error is returned as a string, and when there is a valuestore error the error is returned as json.  If think it would be more consistent to always return the same type of data, especially when the content type header is not being set.

If there is no need to return a parsable structure, then I would prefer to do what is simplest and return a string with a 500 response code in the case of any error, and a 200 in the case of no error.

Reply:
I was trying to distinguish between valuestore and server status but agree this makes it simpler. did changes.